src/components/Main/Main.jsx

```diff
@@ -94,7 +89,7 @@ export function Main({ id, setId, apiToken, setApiToken }) {
     );
 
     return () => clearInterval(timerId);
-  }, [id, apiToken, allMessages, receiptId]);
+  }, [id, apiToken, allMessages, receiptId, number]);
```

Comment:
```suggestion
  }, [id, apiToken, allMessages, receiptId, phoneNumber]);
```

или 

```suggestion
  }, [id, apiToken, allMessages, receiptId, acrivePhoneNumber]);
```


Название number уж слишком общее, как string, не понятно что там в переменной лежит, какие данные